### DISCUSSION TITLE
PR seems to include the wrong changelog and puts everything under "[unreleased]"

### DISCUSSION THREAD

**Discussion by beeb** [release-plz/release-plz#1002] [Q&A] [answered]
I was already using git-cliff for my changelog before migrating my CI to use release-plz today.

I have a cliff.toml config file where I changed some defaults to my liking.

When I get a PR from release-plz, it seems the wrong changelog section (with the old version number) is included, and the Changelog file shows all the changes since last version under the "[unreleased]" section.

https://github.com/beeb/awsbck/pull/36

I would expect the release PR to have the new version number show up at the top of the changelog and all changes under there, as well as that section included in the PR comment/body.

I'm probably doing something wrong but not sure what.

**Comment by marcoieni** [comment, marked as answer]
removing `skip_tag` from `cliff.toml` fixed the issue. Not sure why. The problem is that [this](https://github.com/orhun/git-cliff/blob/baf4da80639682628ca4ae538b4555ff1b6262da/git-cliff-core/src/changelog.rs#L109) is true, and so git-cliff skips the release.

<img width="655" alt="Screenshot 2023-09-30 at 15 25 40" src="https://github.com/MarcoIeni/release-plz/assets/11428655/67a7ba5d-b89d-4e30-8af6-e17d634627c7">

Still doesn't look 100% right, but maybe you can play with other git-cliff options? For example, you might want to remove the footer (I don't think release-plz handles the git-cliff footer correctly), and add `[unreleased]` to the `header`. 


Overall, I should make a better job at documenting how to configure git-cliff for release-plz. See https://github.com/MarcoIeni/release-plz/issues/1006 :/

**Reply by beeb** [reply]
Thanks for the reply and looking into it! Indeed removing the `skip_tag` option helps to find the right section to update, and removing the `footer` also helps. I made those adjustments accordingly.

The only thing that I see is still not right is that the `## [Unreleased]` title is not replaced in the Changelog (as you can see from your screenshot above). I will for now put it inside the header, but a way to handle it being in the body would be nice since all of the example [git-cliff templates](https://github.com/orhun/git-cliff/tree/main/examples) use that conditional.

**Reply by beeb** [reply]
Any idea why the content of the changes for 0.3.5 is now duplicated ? https://github.com/beeb/awsbck/blob/888520131630b79d186a5aa03a8b3d7e4aa96b50/CHANGELOG.md

It looks like release-plz only preprends to the Changelog without considering what was already there. The changelog in `main` was generated with `git cliff -o CHANGELOG.md` locally (so all pending changes are already in the top section, below `[Unreleased]`).

`git cliff` on the other hand re-generates the full changelog it seems.

**Reply by beeb** [reply]
Ideally, release-plz would remove everything above the title for the previous release and re-generate the top portion, or re-generate the whole file just like git-cliff, but pretending there is a new version tag already (so "unreleased" becomes the future version).

The latter would probably be more robust in case of history re-write like amending an older commit or something like that (even though that's probably a rare occurence).

**Reply by marcoieni** [reply]
> Any idea why the content of the changes for 0.3.5 is now duplicated ?

Yes, they were present in the `[unreleased]` section. Delete them and they won't appear again 👍 

To regenerate the whole file, release-plz should analyze the entire git history of the project. At the moment, release-plz only analyzes the latest unpublished changes. However, having the ability to regenerate the whole changelog it's a good idea, but it would take a while to implement. I opened https://github.com/MarcoIeni/release-plz/issues/1009
